### DISCUSSION TITLE
Visual scripting

### DISCUSSION THREAD

**Discussion by Aidedenred6** [godotengine/godot-proposals#6954] [Editor]
Please ad visual scripting in upcoming godot releases godot is a fun engine & would be event better with visual scripting for us who find it difficult to code but live making games & love the engine i have a unity visual scripting plug-in from unity which would love godot devs to implement & expand in godot please let me know where to share the plugin & its writen in c# i want to leave unity fully commit to godot but lack of visual scripting is baring me

**Comment by dalexeev** [comment]
See:

* [Godot 4.0 will discontinue VisualScript](https://godotengine.org/article/godot-4-will-discontinue-visual-scripting/)
* [godotengine/godot-visual-script](https://github.com/godotengine/godot-visual-script)

**Comment by Calinou** [comment]
If you want an open source engine centered around visual scripting, look into [GDevelop](https://gdevelop.io) :slightly_smiling_face:

**Reply by anderlli0053** [reply]
Problem is that it ain't 3D engine :)

**Reply by paddy-exe** [reply]
> Problem is that it ain't 3D engine :)

You can cheat it a little :) https://gdevelop.io/de-de/blog/gdevelop-3d-jam-is-here
